### ISSUE TITLE
Build instructions

### ISSUE THREAD

**Issue by rhyst** [aardgoose/CaveView.js#1]
Hello,

Really interested in the project! Any chance of getting some instructions on how to build it?

Thanks

**Comment by aardgoose** [comment]
Hi,

I can certainly write something up.  Building is node.js rollup based, which aligns well with three.js.
If you haven't used node.js, it's fairly easy to setup/I'm a node novice.

Currently you need a patched version of three.js - I'll make available a fork/branch with the currently required patches, but it mostly works with the three.js 83dev branch.

Terrain support is very custom/OS GB centric, but I intend to sort that out at some point.

Cheers

**Comment by rhyst** [comment]
Ah, thanks! Got it working with rollup.

Had to download your LeakWatch repo (dev branch) and three.js (dev branch) and place them in the same dir as the CaveView.js repo. I also had to change the name of  'src/js/core/colours.js' to 'src/js/core/Colours.js'. (Ubuntu 16.04).

Using it on data from the Migovec system in Slovenia (37km of passage) and it works beautifully. That depth slider is awesome.

**Comment by aardgoose** [comment]
Hi,

That was quick,  the leakwatch is a quick hack to get some internal info out of the webgl renderer to detect leaking GPU memory by accident.  My other three.js branches are either performance tweaks (render-allocs reduces heap churn) or only used when moving from one survey to another in the file menu.  It used to support drag and drop from the desktop, but that is currently broken.

Out of curiosity are you using Survex or Therion models.
Any features you would like to see/change?

**Comment by rhyst** [comment]
Haha, I googled for a bit to try and find something called LeakWatch before checking your other repos :P.

I'm using Survex generated 3d files.

A few things I can think of are (maybe I haven't found them yet):
- Displaying the passage name + station name/number when hovering over nodes
- Centering on a node when clicking it
- Continuous rotate mode (like pressing space in survex) so I've got something to stare at :)

Do you dev on Windows? I'm curious as to why I had to rename that file and am wondering if its a Windows non-case sensitive file name thing.

**Comment by aardgoose** [comment]
Yep, develop mainly on Windows I guess you pulled from master, I fixed the Colour/colour name a while ago. rollup relied on the native platform fs behaviour, eg NTFS broken behaviour, which had interesting issues.  import X from y and import X fromY worked transparently and created two different objects in the combined file which was interesting to debug. I filed an issue with rollup so that was fixed.

Re labeling.  I plan to add that, I now import station names which I had avoided originally.  I think I'll use an HTML overlay to popup info and probably coords (loch displays various details for stations).

I've already got centering and zooming into an individual cave, so centering on a node/station won't be too problematic, stations are now in the survey tree.

The three.js controls i'm using support auto-rotate so that just needs an interface. I partially added it to make it sort of scriptable, eg automatic demo  @ https://aardgoose.github.io/CaveView.js/
